src/components/activities.js

```diff
@@ -251,6 +278,7 @@ export class ActivityPorts {
    * @return {!Promise<!ActivityIframePort>}
    */
   openIframe(iframe, url, opt_args) {
+    opt_args = this.getArgs_(opt_args);
```

Comment:
I think that we need to do a call to AnalyticsService's getContext() before this line because it looks like AnalyticsContext can change throughout the user's journey. At least it seems that way given AnalyticsService's addLabels function.

Reply:
AnalyticsContext is passed by reference so it should always reflect the current object in AnalyticsServer.